types/react-relay/relay-hooks/useMutation.d.ts

```diff
@@ -5,20 +5,45 @@ import {
     IEnvironment,
     MutationConfig,
     MutationParameters,
+    NormalizationOperation,
     PayloadError,
+    ReaderFragment,
+    RequestParameters,
     SelectorStoreUpdater,
     UploadableMap,
     VariablesOf,
 } from 'relay-runtime';
 
+export interface MutationOperation {
+    fragment: ReaderFragment;
+    kind: string;
+    operation: NormalizationOperation;
+    params: RequestParameters;
+    hash: string;
+}
+
+export interface MutationErrorSource<TMutation extends MutationParameters> {
+    errors: PayloadError[];
+    operation: MutationOperation;
+    variables: VariablesOf<TMutation>; // TODO: Does this work correctly?    [name: string]: any;
```

Comment:
good point — not the purpose of this PR. But shall fast follow and double check this. Do you mind spending a min or too, and checking with me?

Reply:
Sure. The `VariablesOf<TMutation>` seems to be working from a type checking perspective. I am not sure why I left a TODO in the code.